src/codeu/chat/util/TransactionLog.java

```diff
@@ -0,0 +1,100 @@
+package codeu.chat.util;
+
+import codeu.chat.server.Controller;
+import codeu.chat.server.Server;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * Utility class for writing and reading to/from the log
+ */
+public class TransactionLog {
+
+  private static final String USER = "ADD-USER";
+  private static final String MESSAGE = "ADD-MESSAGE";
+  private static final String CONVERSATION = "ADD-CONVERSATION";
+  private static final String SPACE = " ";
+
+  public static void writeLog(String type, Object[] params) {
+    String log = "";
+    switch(type) {
+      case "user":
+        log = USER + SPACE +
+                params[0] + SPACE +
+                params[1] + SPACE +
+                params[2];
+        break;
+      case "message":
+        log = MESSAGE + SPACE +
+                params[0] + SPACE +
```

Comment:
using generic object parameters is bad design, since it has made this piece of code very hard to read and understand.